lib/module-index-server/src/routes/reject_module_route.rs

```diff
@@ -107,3 +113,60 @@ pub async fn reject_module(
 
     Ok(Json(Some(response)))
 }
+
+pub async fn reject_other_modules_of_a_schema_id(
+    data: String,
+    schema_id: SchemaId,
+    txn: &sea_orm::DatabaseTransaction,
+) -> RejectModuleResult<()> {
+    let query = si_module::Entity::find();
+
+    // filters
+    let query = query
+        .find_with_linked(SchemaIdReferenceLink)
+        .filter(si_module::Column::SchemaId.eq(schema_id));
+
+    let modules: Vec<ModuleDetailsResponse> = query
+        .all(txn)
+        .await?
+        .into_iter()
+        .map(|(module, linked_modules)| make_module_details_response(module, linked_modules))
+        .collect();
+
+    for module in modules {
+        let module_id = ModuleId::from_str(&module.id)?;
+        let (module, _) = si_module::Entity::find_by_id(module_id)
+            .find_with_linked(si_module::SchemaIdReferenceLink)
+            .all(txn)
+            .await?
+            .first()
+            .cloned()
+            .ok_or(RejectModuleError::NotFound(module_id))?;
+
+        let active_module = si_module::ActiveModel {
+            id: Set(module.id),
+            name: Set(module.name),
+            description: Set(module.description),
+            owner_user_id: Set(module.owner_user_id),
+            owner_display_name: Set(module.owner_display_name),
+            metadata: Set(module.metadata),
+            latest_hash: Set(module.latest_hash),
+            latest_hash_created_at: Set(module.latest_hash_created_at),
+            created_at: Set(module.created_at),
+            rejected_at: Set(Some(DateTime::<FixedOffset>::from_naive_utc_and_offset(
+                Utc::now().naive_utc(),
+                Utc.fix(),
+            ))),
+            rejected_by_display_name: Set(Some(data.clone())),
+            kind: Set(module.kind),
+            is_builtin_at: Set(module.is_builtin_at),
+            is_builtin_at_by_display_name: Set(module.is_builtin_at_by_display_name),
+            schema_id: Set(module.schema_id),
+            ..Default::default() // all other attributes are `NotSet`
+        };
+
+        let _: si_module::Model = dbg!(active_module.update(txn).await)?;
```

Comment:
This dbg should probably not stay (but you also don't need to store the returned value on a variable):
```suggestion
        active_module.update(txn).await?;
```